hal/src/rtl872x/usart_hal.cpp

```diff
@@ -630,15 +698,15 @@ class Usart {
         if (!uart->receiving_) {
             return 0;
         }
-        uart->receiving_ = false;
         auto rxDmaInitStruct = &uart->rxDmaInitStruct_;
-        DCache_Invalidate((uint32_t)rxDmaInitStruct->GDMA_DstAddr, rxDmaInitStruct->GDMA_BlockSize);
         // Clean Auto Reload Bit
 	    GDMA_ChCleanAutoReload(rxDmaInitStruct->GDMA_Index, rxDmaInitStruct->GDMA_ChNum, CLEAN_RELOAD_SRC);
+        GDMA_Cmd(rxDmaInitStruct->GDMA_Index, rxDmaInitStruct->GDMA_ChNum, DISABLE);
         // Clear Pending ISR
 	    GDMA_ClearINT(rxDmaInitStruct->GDMA_Index, rxDmaInitStruct->GDMA_ChNum);
-        GDMA_Cmd(rxDmaInitStruct->GDMA_Index, rxDmaInitStruct->GDMA_ChNum, DISABLE);
-        uart->rxBuffer_.acquireCommit(rxDmaInitStruct->GDMA_BlockSize);
+        UART_RXDMACmd(uartTable_[uart->index_].UARTx, DISABLE);
+        uart->dataInFlight(true /* commit */);
```

Comment:
I think instead we should add `cancel` argument to `dataInFlight()` and cancel-commit the extra buffer size. This way we keep the logic of dealing with transfers in `dataInFlight()` universally.

Reply:
We should not cancel the pending in `dataInFlight()`, cos it might be called to commit data but a DMA transaction is not completed, e.g. in `read()`, unless we add one more argument to differentiate it.

Reply:
Yes, that's the idea to add an argument to differentiate between the two cases but keep the functionality in a single place.